src/core/ext/transport/cronet/transport/cronet_transport.cc

```diff
@@ -1090,22 +1094,21 @@ static enum e_op_result execute_stream_op(struct op_and_state* oas) {
       grpc_slice_buffer write_slice_buffer;
       grpc_slice slice;
       grpc_slice_buffer_init(&write_slice_buffer);
-      if (1 != stream_op->payload->send_message.send_message->Next(
-                   stream_op->payload->send_message.send_message->length(),
-                   nullptr)) {
-        /* Should never reach here */
-        GPR_ASSERT(false);
-      }
-      if (GRPC_ERROR_NONE !=
+      while (1 == stream_op->payload->send_message.send_message->Next(
```

Comment:
The contract for `ByteStream::Next()` is that it should not be called once all bytes have been read from the stream:

https://github.com/grpc/grpc/blob/5c715f2696a5144daf664020ee372db4d2eaf0f0/src/core/lib/transport/byte_stream.h#L48

So instead of checking the result of `Next()` here, this should check to see if there are more bytes still to be read, using something like this:

```
while (write_slice_buffer.length < stream_op->payload->send_message.send_message->length()) {
```

You can then call `Next()` within the body of the loop.

Reply:
BTW, when you do call next, you can assert that it returns true, because currently all send_message ByteStreams have the data synchronously available.  But please add the following comment:

TODO(roth): When we add support for incremental sending, this code will need to be changed to support asynchronous delivery of the send_message payload.

Reply:
done